Blackjack/Game/BlackjackGame.m

```diff
@@ -0,0 +1,254 @@
+}
+
++ (NSInteger)cardsAmountToWin {
+    return 21;
+}
+
++ (NSInteger)dealerMinimumCardEvaluation {
+    return 16;
+}
+
+- (NSArray *)currentActivePlayers {
+    NSMutableArray *activePlayers = [NSMutableArray new];
+    for (Player *player in self.players) {
+        if (player.isPlaying) {
+            [activePlayers addObject:player];
+        }
+    }
+    return activePlayers;
+}
+
+// MARK: - Init
+
+- (instancetype)initWithDeck:(Deck *)deck {
+    self = [super init];
+    if (self) {
+        _deck = deck;
+    }
+    return self;
+}
+
+// MARK: - Game Logic
+
+-(void)setupContestants:(NSInteger) numberOfPlayers {
+    NSMutableArray *newPlayers = [NSMutableArray arrayWithCapacity:numberOfPlayers];
+    
+    for (int i=0; i<numberOfPlayers; i++) {
+        NSString *name = [NSString stringWithFormat:@"Player #%d", i+1];
+        Player *player = [[Player alloc] initWith:name cards:[NSMutableArray new] chips: 50 isPlaying:YES];
+        [newPlayers addObject: player];
+    }
+    self.players = newPlayers;
+    self.currentPlayer = self.players.firstObject;
+}
+
+- (void)setDecision:(Decision)decision {
+    switch (decision) {
+        case HIT: {
+            [self.currentPlayer.cards addObject:[self.deck drawRandomCard:YES]];
+            NSInteger cardsEvaluation = [self evaluateCardsFor:_currentPlayer];
+            [self handlePlayersLogicBasedOn: cardsEvaluation forPlayer: self.currentPlayer];
+            
+            if (!self.currentPlayer.isPlaying) {
+                [self nextPlayer];
+            }
+            break;
+        }
+        case STAND:
+            [self nextPlayer];
+            break;
+            
+        case SURRENDER: {
+            self.dealer.chips += self.currentPlayer.betAmount;
+            self.currentPlayer.betAmount = 0;
+            self.currentPlayer.isPlaying = NO;
+            [self nextPlayer];
+            break;
+        }
+    }
+    [self.delegate updateUI];
+}
+
+- (void)setBet:(NSInteger)amount {
+    if (self.currentPlayer.chips >= amount) {
+        self.currentPlayer.betAmount += amount;
+        self.currentPlayer.chips -= amount;
+        [self nextPlayer];
+        [self.delegate updateUI];
+    }
+}
+
+-(void)nextPlayer {
+    NSArray *currentActivePlayers = [self currentActivePlayers];
+
+    if (currentActivePlayers.count == 0) {
+        [self gameOver];
+        return;
+    }
+    NSInteger index = [self.players indexOfObject: self.currentPlayer];
+
+    if (index < self.players.count - 1) {
+        self.currentPlayer = self.players[index + 1];
+    } else if (index == self.players.count - 1) {
+        
+        [self progressGameState];
+    } else {
+        // TODO: handle case
+    }
+}
+
+// TODO : encapuslate some logic
+-(void)progressGameState {
+    switch (self.gameState) {
+        case COLLECT_BETS:
+            self.gameState = DEAL_CARDS;
```

Comment:
You do not have a break here...
That means that the `DEAL_CARDS` state is unneeded

Reply:
Maybe, but I still want to have a representation of what's going on right now in the game, if someone needs to know.

Reply:
Since you don't do anything with it (and you actually change it 4 lines below) I do not see the reason to do this.

What do you think?